### PR TITLE
Use null propagation operator for windows build

### PR DESCRIPTION
**Pull request by fflaten**
# PR Summary

Replace if not null-checks with null propagation operator to solve IDE0031 with .NET Preview 7.
Additional findings when compiled against win7-x64 (in devcontainer).

## PR Context

Related to #17769

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by fflaten** [comment]
Don't have a Windows-machine to build on atm. but hopefully I caught most of them.

@iSazonov Any tips regarding https://github.com/PowerShell/PowerShell/pull/17795/commits/31408ebc206304b2bf4325cac1cf21025490ed8f ? 
When building in devcontainer with `-Runtime win7-x64` and changing them to `securityDescHandle?.Value.Free();` it failed with:

```
/workspaces/PowerShell/src/System.Management.Automation/engine/remoting/common/RunspaceConnectionInfo.cs(2850,32): error CS1061: 'GCHandle' does not contain a definition for 'Value' and no accessible extension method 'Value' accepting a first argument of type 'GCHandle' could be found (are you missing a using directive or an assembly reference?) [/workspaces/PowerShell/src/System.Management.Automation/System.Management.Automation.csproj]
/workspaces/PowerShell/src/System.Management.Automation/engine/remoting/common/RemoteSessionNamedPipe.cs(507,32): error CS1061: 'GCHandle' does not contain a definition for 'Value' and no accessible extension method 'Value' accepting a first argument of type 'GCHandle' could be found (are you missing a using directive or an assembly reference?) [/workspaces/PowerShell/src/System.Management.Automation/System.Management.Automation.csproj]
```

**Review by PaulHigin — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/engine/remoting/common/RemoteSessionNamedPipe.cs [hunk 1]
```diff
@@ -504,10 +504,12 @@ internal static RemoteSessionNamedPipeServer CreateRemoteSessionNamedPipeServer(
                 securityAttributes);
 
             int lastError = Marshal.GetLastWin32Error();
+#pragma warning disable IDE0031
             if (securityDescHandle != null)
             {
                 securityDescHandle.Value.Free();
             }
+#pragma warning restore IDE0031
```
Comment (fflaten) [inline src/System.Management.Automation/engine/remoting/common/RemoteSessionNamedPipe.cs:512]: Just learned that `?.` will strip of the Nullable from `GCHandle?`-type, but I've never used this combo so want to confirm. Anything wrong with doing this? @PaulHigin @iSazonov 
```suggestion
            securityDescHandle?.Free();
```

Comment (PaulHigin) [reply src/System.Management.Automation/engine/remoting/common/RemoteSessionNamedPipe.cs:512]: Interesting, but that makes sense.  I agree that we should use this change.

**Review by PaulHigin — APPROVED** [review]
LGTM

File: src/System.Management.Automation/engine/remoting/common/RemoteSessionNamedPipe.cs [hunk 2]
```diff
@@ -504,10 +504,7 @@ internal static RemoteSessionNamedPipeServer CreateRemoteSessionNamedPipeServer(
                 securityAttributes);
 
             int lastError = Marshal.GetLastWin32Error();
-            if (securityDescHandle != null)
-            {
-                securityDescHandle.Value.Free();
-            }
+            securityDescHandle?.Free();
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/common/RemoteSessionNamedPipe.cs:507]: It generates more efficient code. I wonder but it is great!
Compere:
[before](https://sharplab.io/#v2:C4LghgzgtgPgAgJgIwFgBQcAMACOSB0ASgK4B2wAllAKb4CS51ATgPYAOAyswG4UDG1CAG50WXAS59iTCsACe+AIJ8BECAGEW5VgBsRaUQGZcCbOuwBvdNhu5jcACzYAsgApNUKFsnTZcgCKCfDJswCxM2BDUUjLygRB8AJSW1rZpAOLqABJgpAAmOtQA/JHRvnFBOfmF2AC82KTEOnqpaTYUAGbYrlExfvF82ACE9Y3Nia1tVmhts9gARnLA1ADaALqlfRUJAELEHR3MdQ3UAO4LS6u95QFB+DsUpGBMcgAy1KQA5sAAFmv6cza11itwS+HS1GADyeLwAYuEoD0yiCBnsDswADTYTCJAGA2zA/qVXIFajHTJVUlKZosPhIragvhow5MLEUkmFAAqcjYtAACo9SNQ8rjJmkAL5i2xS9pdek3AaUmojBpNHQTGb4mzTLWzQnbPhK2gANTAOmItFhTGo1Fcos1WslDuwTvFQA=)
[after](https://sharplab.io/#v2:C4LghgzgtgPgAgJgIwFgBQcAMACOSB0ASgK4B2wAllAKb4CS51ATgPYAOAyswG4UDG1CAG50WXAS59iTCsACe+AIJ8BECAGEW5VgBsRaUQGZcCbOuwBvdNhu5jcACzYAsgApNUKFsnTZcgCKCfDJswCxM2BDUUjLygRB8AJSW1rZpAOLqABJgpAAmOtQA/JHRvnFBOfmF2AC82KTEOnqpaTYUAGbYrlExfvF82ACE9Y3Nia1tVmhts9gARnLA1ADaALqlfRUJAELEHR3MdQ3UAO4LS6u95QFB+DsUpGBMcgAy1KQA5sAAFmv6cza11itwS+HS1GADyeLwAYuEoD0yiCBnsDswADTYTCJAGA2zA/qVXIFajHTJVUlKZosPhIragvhow5MLEUkmFAAqcjYtAACo9SNQ8rjJmkAL5i2xSmyE7Z8SmFIr4WFMajUVyima2SVocVAA===)

**Review by iSazonov — APPROVED** [review]

**Pull request merged**